### PR TITLE
Add a .NET 8.0 TFM

### PR DESCRIPTION
**Pull request by kevinchalet**

### CODE REVIEW

File: src/OpenIddict.EntityFrameworkCore/Stores/OpenIddictEntityFrameworkCoreAuthorizationStore.cs [hunk 1]
```diff
@@ -446,7 +447,11 @@ await foreach (var authorization in authorizations)
                 return null;
             }
 
+#if SUPPORTS_NAVIGATION_ENTRY_LOADING_OPTIONS
```
Comment (kevinchalet) [inline src/OpenIddict.EntityFrameworkCore/Stores/OpenIddictEntityFrameworkCoreAuthorizationStore.cs]: We should be able to revert these changes when bumping EF Core to 8.0 preview7.

File: Directory.Packages.props [hunk 2]
```diff
@@ -341,17 +341,61 @@
...
+    <GlobalPackageReference Include="PolySharp" Condition=" '$(DisablePolySharp)' != 'true' " Version="1.13.1"          />
+  </ItemGroup>
+
+  <!--
+                                          ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
+                                          █████ ▀██ ██ ▄▄▄█▄▄ ▄▄███▀▄▄▀████ ▄▄ ██
+                                          █▀▀██ █ █ ██ ▄▄▄███ █████▀▄▄▀█▀▀█ ▀▄ ██
+                                          █▄▄██ ██▄ ██ ▀▀▀███ █████▄▀▀▄█▄▄█ ▀▀ ██
+                                          ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
+  -->
```
Comment (ilmax) [inline Directory.Packages.props:312]: This is a place where I wouldn't expect to find ascii art 😅

Comment (kevinchalet) [reply Directory.Packages.props:312]: ASCII art FTW! ❤️ 

(wait till you see https://github.com/openiddict/openiddict-core/blob/dev/src/OpenIddict.Client.WebIntegration/OpenIddictClientWebIntegrationProviders.xml 🤣)

Comment (ilmax) [reply Directory.Packages.props:312]: That's awesome 💯

Comment (kevinchalet) [reply Directory.Packages.props:312]: On a more serious note, OpenIddict's `Directory.Packages.props` is quite special, as package versions are defined per-TFM (so that `net6.0` references `Microsoft.Extensions.*` 6.0 for instance): using ASCII art makes identifying the per-TFM sections easier 😃

Comment (ilmax) [reply Directory.Packages.props:312]: I saw that's quite a complex one and now I also know why :)

**Comment by codymullins** [comment]
@kevinchalet what is needed for .NET 8 support? need any assistance?

**Comment by kevinchalet** [comment]
@codymullins we need a newer version of https://github.com/ericsink/SQLitePCL.raw to fix this warning/error:

```
error NETSDK1206: Found version-specific or distribution-specific runtime identifier(s): alpine-arm, alpine-arm64, alpine-x64. Affected libraries: SQLitePCLRaw.lib.e_sqlite3. In .NET 8.0 and higher, assets for version-specific and distribution-specific runtime identifiers will not be found by default. See https://aka.ms/dotnet/rid-usage for details. [/Users/runner/work/openiddict-core/openiddict-core/sandbox/OpenIddict.Sandbox.Console.Client/OpenIddict.Sandbox.Console.Client.csproj::TargetFramework=net8.0
```

An OpenIddict 4.x version explicitly targeting .NET 8.0 will ship in November, once .NET 8.0 itself is available. In the meantime, using OpenIddict 4.8 on .NET 8.0 should work just fine.

**Comment by kevinchalet** [comment]
> need any assistance?

@codymullins thanks for your offer, BTW. To help the project, please don't hesitate to encourage your employer to sponsor it. More info here if you're interested: https://github.com/sponsors/kevinchalet ❤️

**Comment by codymullins** [comment]
@kevinchalet I'm looking at using openiddict as the base for the identity solution I plan to ship with [Pure Blazor](https://pureblazor.com). It's definitely on the list to sponsor this project as soon as possible :-)
